baybe/utils/dataframe.py

```diff
@@ -351,6 +351,48 @@ def df_uncorrelated_features(
     return data
 
 
+def df_add_noise_to_degenerate_rows(
+    df: pd.DataFrame, noise_level: float = 0.001
+) -> pd.DataFrame:
+    """Add noise to degenerate rows to being able to numerically distinguish them.
+
+    Args:
+        df: The dataframe to be modified.
+        noise_level: The magnitude of noise relative to the min-max range of
+            values for each column.
+
+    Returns:
+        The modified dataframe.
+
+    Raises:
+        TypeError: If the provided dataframe has non-numerical content.
+    """
+    # Assert that the input is purely numerical
+    if any(df[col].dtype.kind not in "iufb" for col in df.columns):
+        raise TypeError(
+            f"'{df_add_noise_to_degenerate_rows.__name__}' only supports purely"
```

Comment:
imo not really needed (a simple `.duplciated` call from the user side will provide the same info) and it might overflow the screen if its really many of them

Reply:
Let the two of you decide, but in case you keep it: missing white space at the end of the first line of the error message

Reply:
Ok, then let's keep it as it is. Not resolving yet due to the missing white space.

Reply:
fixed